pkg/golinters/gomoddirectives.go

```diff
@@ -0,0 +1,64 @@
+package golinters
+
+import (
+	"sync"
+
+	"github.com/ldez/gomoddirectives"
+	"golang.org/x/tools/go/analysis"
+
+	"github.com/golangci/golangci-lint/pkg/config"
+	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
+	"github.com/golangci/golangci-lint/pkg/lint/linter"
+	"github.com/golangci/golangci-lint/pkg/result"
+)
+
+const goModDirectivesName = "gomoddirectives"
+
+// NewGoModDirectives returns a new gomoddirectives linter.
+func NewGoModDirectives(settings *config.GoModDirectivesSettings) *goanalysis.Linter {
+	var issues []goanalysis.Issue
+	var once sync.Once
+
+	var opts gomoddirectives.Options
+	if settings != nil {
+		opts.ReplaceAllowLocal = settings.ReplaceLocal
+		opts.ReplaceAllowList = settings.ReplaceAllowList
+		opts.RetractAllowNoExplanation = settings.RetractAllowNoExplanation
+		opts.ExcludeForbidden = settings.ExcludeForbidden
```

Comment:
if you use a custom type or line up the structs, you can do a direct assignment:
```suggestion
		opts = gomoddirectives.Options(settings)
```

Reply:
it's not the same type, so it's not possible.

Reply:
actually, it is: https://play.golang.org/p/Gna-Uv1PAaS

Reply:
```
# github.com/golangci/golangci-lint/pkg/golinters
pkg/golinters/gomoddirectives.go:24:33: cannot convert *settings (type "github.com/golangci/golangci-lint/pkg/config".GoModDirectivesSettings) to type gomoddirectives.Options
```

Reply:
https://play.golang.org/p/0PbSOa7_aI4

Reply:
you changed the name of `ReplaceAllowLocal`: https://play.golang.org/p/o-97b9emRD_X